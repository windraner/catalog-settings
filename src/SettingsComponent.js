import React, { Component } from 'react';
import Title from './components/title/Title';
import SettingsContent from './components/settingsContent/SettingsContent';
import ButtonCancel from './components/buttonCancel/ButtonCancel';
import ButtonSave from './components/buttonSave/ButtonSave';
import PropTypes from 'prop-types';

import './catalogSettings.css';

export default class SettingsComponent extends Component {
  state = {
    newData: {}
  }

  getFormData = (data) => {
    this.setState({newData: data});
  }

  onSaveCallback = () => {
    const event = new Event('SendForm');
    window.dispatchEvent(event);
    setTimeout(() => {
      const {title, defaultLanguage} = this.state.newData;

      if(!title || !defaultLanguage) {return;}

      this.props.onSave(this.state.newData);
    }, 0);
  }

  onCancelCallback = () => {
    this.props.onCancel();
  }

  render() {
    const { data, flags, languages } = this.props;

    return (
      <div>
        <Title />

        <SettingsContent
          data={data}
          flags={flags}
          languages={languages}
          getFormData={this.getFormData}
        />

        <div className="catalog-settings__splitter margin-left-auto margin-right-auto margin-top-30 margin-bot-30">
          <ButtonCancel
            handlerCallBack={this.onCancelCallback}
          />
          <ButtonSave
            handlerCallBack={this.onSaveCallback}
          />
        </div>
      </div>
    );
  }
}

SettingsComponent.propTypes = {
  data: PropTypes.object.isRequired,
  flags: PropTypes.object.isRequired,
  languages: PropTypes.array.isRequired,
  onSave: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
};
