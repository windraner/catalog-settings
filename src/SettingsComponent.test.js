import React from 'react';
import SettingsComponent from './SettingsComponent';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() })

const data = {
  title: 'Catalog title',
  description: 'Catalog description',
  defaultLanguage: 'en',
  additionalLanguages: ['es', 'ru'],
  defaultFallback: true,
  fileUrl: '',
  file: '',
  hideVariants: false
};

const props = {
  data,
  flags: {
    canUploadFile: true,
    canHideVariants: true
  },
  languages: ['en', 'es', 'ru', 'de'],
  onSave: function() {},
  onCancel: function() {},
}

describe('<SettingsComponent />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<SettingsComponent {...props} />)
    expect(tree).toMatchSnapshot()
  });
});
