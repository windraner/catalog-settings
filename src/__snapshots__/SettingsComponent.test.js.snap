// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsComponent /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SettingsComponent
    data={
      Object {
        "additionalLanguages": Array [
          "es",
          "ru",
        ],
        "defaultFallback": true,
        "defaultLanguage": "en",
        "description": "Catalog description",
        "file": "",
        "fileUrl": "",
        "hideVariants": false,
        "title": "Catalog title",
      }
    }
    flags={
      Object {
        "canHideVariants": true,
        "canUploadFile": true,
      }
    }
    languages={
      Array [
        "en",
        "es",
        "ru",
        "de",
      ]
    }
    onCancel={[Function]}
    onSave={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Title />,
        <SettingsContent
          data={
            Object {
              "additionalLanguages": Array [
                "es",
                "ru",
              ],
              "defaultFallback": true,
              "defaultLanguage": "en",
              "description": "Catalog description",
              "file": "",
              "fileUrl": "",
              "hideVariants": false,
              "title": "Catalog title",
            }
          }
          flags={
            Object {
              "canHideVariants": true,
              "canUploadFile": true,
            }
          }
          getFormData={[Function]}
          languages={
            Array [
              "en",
              "es",
              "ru",
              "de",
            ]
          }
        />,
        <div
          className="catalog-settings__splitter margin-left-auto margin-right-auto margin-top-30 margin-bot-30"
        >
          <ButtonCancel
            handlerCallBack={[Function]}
          />
          <ButtonSave
            handlerCallBack={[Function]}
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "additionalLanguages": Array [
              "es",
              "ru",
            ],
            "defaultFallback": true,
            "defaultLanguage": "en",
            "description": "Catalog description",
            "file": "",
            "fileUrl": "",
            "hideVariants": false,
            "title": "Catalog title",
          },
          "flags": Object {
            "canHideVariants": true,
            "canUploadFile": true,
          },
          "getFormData": [Function],
          "languages": Array [
            "en",
            "es",
            "ru",
            "de",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ButtonCancel
              handlerCallBack={[Function]}
            />,
            <ButtonSave
              handlerCallBack={[Function]}
            />,
          ],
          "className": "catalog-settings__splitter margin-left-auto margin-right-auto margin-top-30 margin-bot-30",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handlerCallBack": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handlerCallBack": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Title />,
          <SettingsContent
            data={
              Object {
                "additionalLanguages": Array [
                  "es",
                  "ru",
                ],
                "defaultFallback": true,
                "defaultLanguage": "en",
                "description": "Catalog description",
                "file": "",
                "fileUrl": "",
                "hideVariants": false,
                "title": "Catalog title",
              }
            }
            flags={
              Object {
                "canHideVariants": true,
                "canUploadFile": true,
              }
            }
            getFormData={[Function]}
            languages={
              Array [
                "en",
                "es",
                "ru",
                "de",
              ]
            }
          />,
          <div
            className="catalog-settings__splitter margin-left-auto margin-right-auto margin-top-30 margin-bot-30"
          >
            <ButtonCancel
              handlerCallBack={[Function]}
            />
            <ButtonSave
              handlerCallBack={[Function]}
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "additionalLanguages": Array [
                "es",
                "ru",
              ],
              "defaultFallback": true,
              "defaultLanguage": "en",
              "description": "Catalog description",
              "file": "",
              "fileUrl": "",
              "hideVariants": false,
              "title": "Catalog title",
            },
            "flags": Object {
              "canHideVariants": true,
              "canUploadFile": true,
            },
            "getFormData": [Function],
            "languages": Array [
              "en",
              "es",
              "ru",
              "de",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ButtonCancel
                handlerCallBack={[Function]}
              />,
              <ButtonSave
                handlerCallBack={[Function]}
              />,
            ],
            "className": "catalog-settings__splitter margin-left-auto margin-right-auto margin-top-30 margin-bot-30",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handlerCallBack": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handlerCallBack": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
